### PR TITLE
Remove `AggregateCommandScope.Create()` and `Exists().

### LINKED ISSUE #127: Consider allowing Root() to be called for aggregate/process instances that do not exist.
This issue proposes changing `Root()` to return value from the handler's `New()` method when the instance does not exist (instead of panicking, as it currently does).

The mechanical benefit of doing so is that we could then call methods on the root to handle the message without having to check `Exists()` first, but more importantly it allows use of domain-level logic to determine how to respond to the command, rather than engine/handler level "instance existence" checks.

### PR DESCRIPTION
**Pull request by jmalloc**
Fixes #127

### CODE REVIEW

File: docs/adr/0016-automatic-aggregate-creation.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+From the handler implementors perspective there are numerous subtle interactions
+between the aggregate instance's "state of existence" and the domain logic.
+
+Perhaps most egregiously is the requirement that an instance already exist
+before calling `Root()`. This means that the state of the aggregate root itself
+can not be used to determine which events should be recorded without first
+creating the instance. A less obvious problem is that when the domain logic is
+such that `RecordEvent()` is not called within the same scope as a call to
+create `Create()` the engine must panic.
```
Comment (koden-km) [inline docs/adr/0016-automatic-aggregate-creation.md:28]: Is this an extra _create_?
```suggestion
`Create()` the engine must panic.
```

File: docs/adr/0016-automatic-aggregate-creation.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+### Effect on engine implementations
+
+From the engine implementors perspective it seems that a non-trivial amount of
+validation logic needs to be implemented within the scope to verify that the
+application code is using the scope correctly, even though the engine does not
+really benefit from this requirements.
+
+For example, event sourced engine implementations tend to call
+`AggregateMessageHandler.New()` to construct a new instance before apply
```
Comment (koden-km) [inline docs/adr/0016-automatic-aggregate-creation.md:47]:
```suggestion
`AggregateMessageHandler.New()` to construct a new instance before applying
```

File: docs/adr/0016-automatic-aggregate-creation.md (hunk 1 above)
Comment (jmalloc) [reply docs/adr/0016-automatic-aggregate-creation.md:28]: Yep, ta.

File: docs/adr/0016-automatic-aggregate-creation.md (hunk 2 above)
Comment (koden-km) [reply docs/adr/0016-automatic-aggregate-creation.md:47]: Somehow this second change suggestion didn't get added to the first review response.

**Review by koden-km — APPROVED** [review]

File: docs/adr/0016-automatic-aggregate-creation.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+historical events via `ApplyEvent()`. This means that the root instance is
+constructed in memory before `HandleCommand()` is even invoked. The requirement
+that `Root()` panic if the instance does not exist does not save the engine from
```
Comment (danilvpetrov) [inline docs/adr/0016-automatic-aggregate-creation.md:50]:
```suggestion
that `Root()` panics if the instance does not exist does not save the engine from
```

File: docs/adr/0016-automatic-aggregate-creation.md (hunk 2 above)
Comment (danilvpetrov) [inline docs/adr/0016-automatic-aggregate-creation.md]:
```suggestion
`AggregateMessageHandler.New()` to construct a new instance before applying
```

File: docs/adr/0016-automatic-aggregate-creation.md (hunk 3 above)
Comment (jmalloc) [reply docs/adr/0016-automatic-aggregate-creation.md:50]: I don't know how to explain this, but `panic` sounds right there to me.  It's the prefix "The requirement that" which makes it so. If I ditch that then `panics` sounds right.

File: docs/adr/0016-automatic-aggregate-creation.md (hunk 2 above)
Comment (jmalloc) [reply docs/adr/0016-automatic-aggregate-creation.md]: This one is fixed already :D

File: docs/adr/0016-automatic-aggregate-creation.md (hunk 3 above)
Comment (danilvpetrov) [reply docs/adr/0016-automatic-aggregate-creation.md:50]: Yep, seems to be clear now.

**Review by danilvpetrov — APPROVED** [review]

**Pull request merged**